### PR TITLE
Adds a document regarding the usage of storage commands.

### LINKED ISSUE #3761: Add documentation of storage commands for devfile components
/kind documentation

## What is the relevance of it?

We need documentation of storage commands for devfile components once the PRs are merged.

- [ ] `odo storage create`
- [ ] `odo storage delete`
- [ ] `odo storage list`

### PR DESCRIPTION
**Pull request by mik-dass**
**What type of PR is this?**

/kind documentation

**What does does this PR do / why we need it**:

It adds a document about the usage of storage commands.

**Which issue(s) this PR fixes**:

Fixes #3761

**PR acceptance criteria**:

- [ ] Unit test 

- [ ] Integration test 

- [X] Documentation 

- [X] I have read the [test guidelines](https://github.com/openshift/odo/blob/master/docs/dev/test-architecture.adoc)

**How to test changes / Special notes to the reviewer**:

Check if the document is correct, clear and easy to understand.

### CODE REVIEW

**Comment by mik-dass** [comment]
```
level=error msg="Attempted to gather ClusterOperator status after installation failure: listing ClusterOperator objects: Get https://api.ci-op-i335nzk0-2f74e.origin-ci-int-aws.dev.rhcloud.com:6443/apis/config.openshift.io/v1/clusteroperators: dial tcp 3.219.116.198:6443: connect: connection refused"
```

**Comment by mik-dass** [comment]
```
level=error msg="Attempted to gather ClusterOperator status after installation failure: listing ClusterOperator objects: Get https://api.ci-op-i335nzk0-2f74e.origin-ci-int-aws.dev.rhcloud.com:6443/apis/config.openshift.io/v1/clusteroperators: dial tcp 3.219.116.198:6443: connect: connection refused"
```
/retest

**Comment by cdrage** [comment]
Hey @mik-dass unfortunately the document isn't the same format as other documentation such as: https://github.com/openshift/odo/blob/b0cc651f8f14ec7e17ef38d33b0fe234e637b85b/docs/public/deploying-a-devfile-using-odo.adoc

There are certain (unwritten) requirements that we follow from the OpenShift docs team. @boczkowska unsure if Yana has an official link to it.

For example, for each command, we need to have `$` in front of the command, and for anything that's ran by root, it has to have `#`

File: docs/public/using-storage.adoc [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# Storage commands
+
+`odo` lets users manage storage in the cluster. A storage is a https://kubernetes.io/docs/concepts/storage/volumes/#persistentvolumeclaim[PVC], which is a way for users to "claim" durable storage (such as a GCE PersistentDisk or an iSCSI volume) without knowing the details of the particular cloud environment.
+
+# Adding a storage
+
+We can add a storage to the cluster using `odo storage create`
+
+```
```
[the remark this answers is not in the document]
Comment (cdrage) [reply docs/public/using-storage.adoc:9]: Have to use: 

```
. This is an example heading:
+
[source,sh]
----
```

**Review by cdrage — CHANGES_REQUESTED** [review]
Have to write this in adoc!

**Comment by mik-dass** [comment]
@cdrage Fixed

**Comment by girishramnani** [comment]
I am thinking about `you can a storage` - I mean shouldn’t it be “storage volume”. And also should we explain what that means? Because a new user would naturally ask that “without “storage” I don’t have any storage capacity in my component?”
Explaining that adding a storage volumes means that data in that directory would persist across rebuilds seems like a good idea. Just a suggestion.

**Comment by mik-dass** [comment]
@girishramnani Fixed.

**Comment by kadel** [comment]
This is good start.
/lgtm
/approve

**Comment by kadel** [comment]
/override ci/prow/v4.5-integration-e2e 
this is just docs, we don't need to wait for tests

**Pull request merged**